src/transform-array.js

```diff
@@ -1,6 +1,44 @@
 const CustomError = require("../extensions/custom-error");
 
-module.exports = function transform(/* arr */) {
-  throw new CustomError('Not implemented');
-  // remove line with error and write your code here
+module.exports = function transform(arrayToBeTransformed) {
+  if(!Array.isArray(arrayToBeTransformed)){
+    throw new CustomError('It is not array!');
+  }
+  let result_array = arrayToBeTransformed;
+
+  for (let i=0; i<arrayToBeTransformed.length; i++){
+    if(typeof(arrayToBeTransformed[i]) == 'string'){
+      switch(arrayToBeTransformed[i]){
+        case '--discard-prev':
+          result_array.splice(i,1);
+          if(i!==0){
+            result_array.splice(i-1,1);}
+          break;
+        case '--discard-next':
+          if(i!==arrayToBeTransformed.length-1){
+            result_array.splice(i+1,1);
+          }
+          result_array.splice(i,1);
+          break;
+        case '--double-next':
+          if(i!==arrayToBeTransformed.length-1){
+            result_array.splice(i, 1, arrayToBeTransformed[i+1]);
+          }
+          else {
+            result_array.splice(i,1);
+          }
+          break;
+        case '--double-prev':
+          if(i!==0){
+            result_array.splice(i, 1, arrayToBeTransformed[i-1]);
+          }
+          else {
+            result_array.splice(i,1);
+          }
+          break;
+
+      }
+    }
+  }
```

Comment:
I'd use reduce (yes, again) instead 'for' loop. Also instead of switcher you could use object with keys like '--double-next' and value will be function for handling this case, which takes current index, current array and initial array.